libarchive/archive_write_disk_posix.c

```diff
@@ -4633,7 +4633,7 @@ set_xattrs(struct archive_write_disk *a)
 				e = extattr_set_fd(a->fd, namespace, name,
 				    value, size);
 				if (e == 0 && errno == 0) {
-					e = size;
+					e = (int)size;
```

Comment:
Why was this cast added?

Reply:
Hmm.  `size` is a `size_t`, so a compiler might give a warning about assigning that value to an `int`.  (I haven't enabled `-Wshorten-64-to-32` yet, so I haven't run into that warning yet.)

On line 4643, the pre-existing code has
```
if (e != (int)size) {
```
so the cast isn't crazy idea.

However, I don't think the cast belongs in a commit with that subject line.  And more generally, that whole section is a mess regarding types.  `int e` is also assigned the return value from `ssize_t extattr_set_link()` on line 4639, which is another unsafe (implicit) cast.